src/BarChart/src/features/barchart/Board.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import styled from "styled-components";
+import Ruler from "./Ruler";
+import Detail from "./Detail";
+import Level from "./Level";
+
+const ZoomInOut = styled.div`
+    width: ${(props) => props.ratio || 100}%;
+`;
+
+class Board extends Component {
+    constructor(props) {
+        super(props);
+        this.handleRulerCntClick = this.handleRulerCntClick.bind(this);
+        this.handleRulerCntMultipleClick = this.handleRulerCntMultipleClick.bind(this);
+        this.clickBar = this.clickBar.bind(this);
+        this.openFileSelector = this.openFileSelector.bind(this);
+        this.processFile = this.processFile.bind(this);
+        this.processData = this.processData.bind(this);
+    }
+
+    state = {
+        rulerCnt: 6,
+        ratio: 100,
+        selectedOP: null,
+        fileName: null,
+        data: null,
+        MaxEndTime: null,
+        utility: null
+    }
+
+    handleRulerCntClick(value){
+        this.setState({ratio: this.state.ratio + value})
+    }
+
+    handleRulerCntMultipleClick(value){
+        this.setState({ratio: this.state.ratio * value})
+    }
+
+    clickBar(info){
+        this.setState({selectedOP: info})
+    }
+
+    openFileSelector(){
+        const input = document.createElement("input");
+        input.type = "file";
+        input.accept = "text/plain";
+        input.onchange = (event) => {
+            this.setState({fileName: event.target.files[0].name})
+            this.processFile(event.target.files[0]);
+        };
+        input.click();
+    }
+
+    processFile(file) {
+        const reader = new FileReader();
+        reader.onload = () => {
+            const data = JSON.parse(reader.result).traceEvents
+            this.processData(data)
+        }
+        reader.readAsText(file, /* optional */ "euc-kr");
+    }
+
+    processData(data) {
+        const processedData = {}
+        const backgroundColor = {}
+        const utility = {"CPU Operators": 0, "GPU Operators": 0}
```

Comment:
`CPU Operators` and `GPU operators` are not fixed name :(
It should be read from json file...

Reply:
all right! Then I'll fix it to show % in all pids! : )